tests/unit/button/button_options.js

```diff
@@ -25,135 +25,80 @@ test( "disabled, explicit value", function() {
-	deepEqual(true, $("#radio02").button("option", "disabled"),
-		"disabled option set to true");
-	deepEqual(true, $("#radio02").prop("disabled"), "element is not disabled");
-});
-
-test( "disabled, ui-state-active is removed unless checkbox or radio", function() {
-	expect( 12 );
-	var elements = [
-		$( "<input type='button'>" ),
-		$( "<button></button>" ),
-		$( "<a></a>" ),
-		$( "<div></div>" ),
-		$( "<input type='checkbox' id='checkbox' checked><label for='checkbox'></label>" ),
-		$( "<input type='radio' id='radio' checked><label for='radio'></label>" )
-	];
-
-	$.each( elements, function() {
-		var element = $( this ).first().button(),
-			buttonElement = element.button( "widget" ),
-			elementType = element.prop( "nodeName" ).toLowerCase();
-
-		if ( element.is( "input" ) ) {
-			elementType += ":" + element.attr( "type" );
-		}
-
-		element.trigger( "mousedown" );
-		ok( buttonElement.hasClass( "ui-state-active" ),
-			"[" + elementType + "] has ui-state-active class after mousedown." );
-
-		element.button( "disable" );
-		if ( element.is( "[type=checkbox], [type=radio]" ) ) {
-			ok( buttonElement.hasClass( "ui-state-active" ),
-				"Disabled [" + elementType + "] has ui-state-active class." );
-		} else {
-			ok( !buttonElement.hasClass( "ui-state-active" ),
-				"Disabled [" + elementType + "] does not have ui-state-active class." );
-		}
-	});
+	strictEqual( $("#button").prop("disabled"), false, "element is disabled");
 });
 
-test("text false without icon", function() {
+test( "showLabel false without icon", function() {
 	expect( 1 );
-	$("#button").button({
-		text: false
+	$( "#button" ).button({
+		showLabel: false
 	});
-	ok( $("#button").is(".ui-button-text-only:not(.ui-button-icon-only)") );
+	strictEqual( $( "#button" ).is( ":ui-button.ui-corner-all.ui-widget" ), true,
+		"Button has correct classes" );
 
-	$("#button").button("destroy");
+	$( "#button" ).button( "destroy" );
 });
 
-test("text false with icon", function() {
+test("showLabel false with icon", function() {
 	expect( 1 );
 	$("#button").button({
-		text: false,
-		icons: {
-			primary: "iconclass"
-		}
+		showLabel: false,
+		icon: "iconclass"
 	});
-	ok( $("#button").is(".ui-button-icon-only:not(.ui-button-text):has(span.ui-icon.iconclass)") );
-
-	$("#button").button("destroy");
+	strictEqual( $( "#button" ).is( ".ui-button.ui-corner-all.ui-widget.ui-button-icon-only" ),
+		true, "Button has correct classes" );
+	$( "#button" ).button( "destroy" );
 });
 
-test("label, default", function() {
+test( "label, default", function() {
 	expect( 2 );
-	$("#button").button();
-	deepEqual( $("#button").text(), "Label" );
-	deepEqual( $( "#button").button( "option", "label" ), "Label" );
+	$( "#button" ).button();
+	deepEqual( $( "#button" ).text(), "Label" );
+	deepEqual( $( "#button" ).button( "option", "label" ), "Label" );
 
-	$("#button").button("destroy");
+	$( "#button" ).button( "destroy" );
 });
 
-test("label", function() {
+test( "label", function() {
```

Comment:
Let's make these names more explicit. I suggest this to be "label, explicit value"